rave-init.js

```diff
@@ -0,0 +1,50 @@
+// TODOs:
+// add a loader on clicking pay now. takes some secs before the rave modal shows up
+// Other use cases: Covered shopping cart for now
+
+require('core-js/es6/promise');
+var Joi = require('joi');
+var get = require('lodash/get');
+var partial = require('lodash/partial');
+var merge = require('lodash/merge');
+
+var raveConfig = require('./config.json');
+var reqFields = require('./schemas').dynamicFields;
+
+function initRavePay(email, amount, onclose, callback) {
+  Promise.resolve(raveConfig)
+    .then(partial(mergeReqUserFields, amount, email))
+    .then(ensureRequired)
+    .then(partial(showModal, onclose, callback))
+    .catch(err => {
+      var errorText = document.createElement('p');
```

Comment:
Consider extracting the DOM manipulation to a separate function that can easily mocked when testing.